azure-quantum/azure/quantum/workspace.py

```diff
@@ -218,6 +248,40 @@ def user_agent(self):
             full_user_agent = f"{full_user_agent}-{env_app_id}" if full_user_agent else env_app_id
         return full_user_agent
 
+    @classmethod
+    def _parse_connection_string(cls, connection_string: str):
+        # The connection string looks like:
+        #SubscriptionId=<subId>;ResourceGroupName=<resourceGroupName>;WorkspaceName=<workspaceName>;
+        #ApiKey=<workspacekey>;QuantumEndpoint=https://<location>.quantum.azure.com/; 
+        regex = r"SubscriptionId=([a-fA-F0-9-]*);ResourceGroupName=([^\s/]*);WorkspaceName=([^\s/]*);ApiKey=([^\s/]*);QuantumEndpoint=https://([^\s/]*).quantum.azure.com/;"  
+        match = re.search(regex, connection_string, re.IGNORECASE)
+        if match:
+            # match should contain four groups:
+            # -> match.group(0):
+            # The full resource ID for the Azure Quantum workspace
+            # -> match.group(1): The Azure subscription ID
+            # -> match.group(2): The Azure resource group name
+            # -> match.group(3): The Azure Quantum workspace name
+            subscription_id = match.group(1)
+            resource_group = match.group(2)
+            workspace_name = match.group(3)
+            workspace_key = match.group(4)
+            location = match.group(5)
+
+            return subscription_id, resource_group, workspace_name, location, workspace_key
+
+        raise Exception("Invalid connection string")
+
+    @classmethod
+    def from_connection_string(cls, connection_string, **kwargs):
```

Comment:
please type out the kwargs that are accepted here, similar to the ones in the constructor

Reply:
also docstrings should be here